src/v2/api/services/ScheduleService.ts

```diff
@@ -237,4 +239,55 @@ export class ScheduleService {
       discipline: result.discipline,
     };
   }
+
+  async getAllGroupEvents (id, week) {
+    const { startOfWeek, endOfWeek } = week ? await this.dateService.getDatesOfWeek(week) : this.dateService.getDatesOfCurrentWeek();
+    const events = await this.eventRepository.findMany({
+      where: {
+        groupId: id,
+        endTime: {
+          gte: startOfWeek,
+        },
+        startTime: {
+          lte: endOfWeek,
+        },
+      },
+    }) as unknown as DbEvent[];
+
+    return events
+      .filter((event) => {
+        const indexOfLesson = this.getIndexOfLesson(event.period, event.startTime, endOfWeek);
+        return indexOfLesson !== null;
+      });
+  }
+
+  async getGroupEvents (
+    userId: string,
+    groupId: string,
+    isOwnSelected: boolean,
+  ) {
+    const user = await this.userService.getUser(userId);
+    const isUserGroup = user.group.id === groupId;
+    let groupEvents: DbEvent[];
+    if (isUserGroup) {
+      groupEvents = await this.getAllGroupEvents(groupId, null);
+    } else {
+      return await this.getGeneralGroupEvents(groupId, null);
+    }
+
+    if (isOwnSelected) {
+      const userSelective = await this.userService.getSelectiveDisciplines(userId);
+      await filterAsync(groupEvents,  async (event) => {
+        const lesson = event.lessons[0];
+        if ((await this.disciplineRepository.findById(lesson.disciplineType.disciplineId)).isSelective) {
+          return userSelective.some((selectiveDiscipline) =>
+            selectiveDiscipline.id === lesson.disciplineType.disciplineId);
+        } else {
+          return true;
```

Comment:
making else in a such way is a bad practice. you can revert your if-statement, making it more readable:
```ts
const discipline = ...;
if (!discipline.isSelective) return true;

return some(...);
```